source/jormungandr/jormungandr/schedule.py

```diff
@@ -0,0 +1,94 @@
+# coding=utf-8
+
+# Copyright (c) 2001-2014, Canal TP and/or its affiliates. All rights reserved.
+#
+# This file is part of Navitia,
+#     the software to build cool stuff with public transport.
+#
+# Hope you'll enjoy and contribute to this project,
+#     powered by Canal TP (www.canaltp.fr).
+# Help us simplify mobility and open public transport:
+#     a non ending quest to the responsive locomotion way of traveling!
+#
+# LICENCE: This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+#
+# Stay tuned using
+# twitter @navitia
+# IRC #navitia on freenode
+# https://groups.google.com/d/forum/navitia
+# www.navitia.io
+
+from navitiacommon import type_pb2, request_pb2
+from jormungandr.utils import date_to_timestamp
+
+
+class MixedSchedule(object):
+
+    def __init__(self, instance):
+        self.instance = instance
+
+    def __stop_times(self, request, api, departure_filter="", arrival_filter=""):
+        req = request_pb2.Request()
+        req.requested_api = api
+        st = req.next_stop_times
+        st.departure_filter = departure_filter
+        st.arrival_filter = arrival_filter
+        if request["from_datetime"]:
+            st.from_datetime = request["from_datetime"]
+        if request["until_datetime"]:
+            st.until_datetime = request["until_datetime"]
+        st.duration = request["duration"]
+        st.depth = request["depth"]
+        st.show_codes = request["show_codes"]
+        if "nb_stoptimes" not in request:
+            st.nb_stoptimes = 0
+        else:
+            st.nb_stoptimes = request["nb_stoptimes"]
+        if "interface_version" not in request:
+            st.interface_version = 0
```

Comment:
not mandatory part of your PR, but I think this interface_version is not useful anymore